llvm/include/llvm/Support/GenericLoopInfo.h

```diff
@@ -75,6 +77,13 @@ template <class BlockT, class LoopT> class LoopBase {
   const LoopBase<BlockT, LoopT> &
   operator=(const LoopBase<BlockT, LoopT> &) = delete;
 
+  // Induction variable exit value and its debug users, preserved by the
+  // 'indvars' pass, when it detects that the loop can be deleted and the
+  // there are no PHIs to be rewritten.
+  // For now, we only preserve single induction variables.
+  Value *IndVarFinalValue = nullptr;
+  SmallVector<WeakVH> IndVarDebugUsers;
```

Comment:
Why is it necessary for these to be members on the Loop object itself?

Reply:
It also introduced some link errors. Reverting the patch.

Reply:
Thanks for reverting. I will look at that layering violation.